v2/api/insights/v1api20230311/structure.txt

```diff
@@ -0,0 +1,156 @@
+// Code generated by azure-service-operator-codegen. DO NOT EDIT.
+github.com/Azure/azure-service-operator/v2/api/insights/v1api20230311
+---------------------------------------------------------------------
+APIVersion: Enum (1 value)
+└── "2023-03-11"
+
+DataCollectionEndpoint: Resource
+├── Owner: resources/v1apiv20191001.ResourceGroup
+├── Spec: Object (10 properties)
+│   ├── AzureName: string
+│   ├── Description: *string
+│   ├── Identity: *Object (2 properties)
+│   │   ├── Type: *Enum (4 values)
+│   │   │   ├── "None"
+│   │   │   ├── "SystemAssigned"
+│   │   │   ├── "SystemAssigned,UserAssigned"
+│   │   │   └── "UserAssigned"
+│   │   └── UserAssignedIdentities: Object (1 property)[]
+│   │       └── Reference: genruntime.ResourceReference
+│   ├── ImmutableId: *string
```

Comment:
I assume that this is not an ARM ID.

Is this user configurable?

Reply:
Correct it's not an ARM ID. AFAIK it's generated by the service and not usable settable, I'll remove it from the Spec.